src/views/OtherStoresInventoryModal.vue

```diff
@@ -58,38 +112,192 @@ export default defineComponent({
     IonNote,
     IonSearchbar,
     IonTitle,
-    IonToolbar 
+    IonToggle,
+    IonToolbar,
+    IonSpinner
   },
-  props: ["otherStoresInventory"],
+  props: ["otherStoresInventory", "currentFacilityId"],
   data() {
-    return{
+    return {
       queryString: "",
-      storesInventory: [] as any
+      isRefreshing: false,
+      storesInventory: [] as any, // will be used when fallback
+      storesWithInventory: [] as any, // will be used primarily.
+      hideEmptyStores: false,
+      activeSortType: "name", // 'name' or 'distance'
+      weekArray: ["sunday", "monday", "tuesday", "wednesday", "thursday", "friday", "saturday"]
     }
   },
-  mounted() {
+  computed: {
+    ...mapGetters({
+      FacilityInformation: 'util/getCurrentFacilityInformation',
+      storesInformation: 'util/getStoresInformation'
+    }),
+    currentFacilityCoords() {
+      return this.FacilityInformation;
+    },
+    nearbyStores() {
+      return this.storesInformation || [];
+    },
+  },
+  async mounted() {
+    this.isRefreshing = true;
     // Create a copy of otherStoresInventory on mount
     this.storesInventory = this.otherStoresInventory.slice();
+    // Fetching Lat Lon for current facility and then fetching store lookup
+    try {
+      await this.store.dispatch("util/fetchCurrentFacilityInformation", this.currentFacilityId);
+      if (this.currentFacilityCoords?.latitude && this.currentFacilityCoords?.longitude) {
+        await this.store.dispatch("util/fetchStoresInformation", {
+          latitude: this.currentFacilityCoords.latitude,
+          longitude: this.currentFacilityCoords.longitude
+        });
```

Comment:
We should not fetch facility latLng and stores information on every mounted hook call, as this information is not changed frequently.
We should only call this apis if the state does not have data for these store states, and clear this data whenever facility changes from settings page.